common/src/main/java/com/wynntils/features/user/overlays/MiniMapOverlayFeature.java

```diff
@@ -136,6 +142,80 @@ private void rotateMapToPlayer(PoseStack poseStack, float centerX, float centerZ
             poseStack.translate(-centerX, -centerZ, 0);
         }
 
+        // TODO move most of the buffer builder code into RenderUtils, maybe separate each map border into different
+        // files to avoid insanity involving tx1/ty1/tx2/ty2 + remove magic constant
+        private void renderRectangularMapBorder(
+                PoseStack poseStack, float renderX, float renderY, float width, float height) {
+            Texture texture = null;
+            int grooves = 0;
+            int tx1 = 0;
+            int ty1 = 0;
+            int tx2 = 0;
+            int ty2 = 0;
+
+            switch (borderType) {
+                case Wynn -> {
+                    texture = Texture.WYNN_MAP_TEXTURES;
+                    grooves = 3;
+
+                    tx2 = 112;
+                    ty2 = 112;
+                }
+                case Gilded -> {
+                    texture = Texture.GILDED_MAP_TEXTURES;
+                    grooves = 1;
+
+                    ty1 = 262; // not 0 for some reason
+                    tx2 = 262;
+                    ty2 = 524;
+                }
+                case Paper -> {
+                    texture = Texture.PAPER_MAP_TEXTURES;
+                    grooves = 3;
+
+                    tx2 = 217;
+                    ty2 = 217;
+                }
+            }
+
+            float uScale = 1f / texture.width();
+            float vScale = 1f / texture.height();
+
+            // Scale to stay the same.
+            float groovesWidth = grooves * width / DEFAULT_SIZE;
+            float groovesHeight = grooves * height / DEFAULT_SIZE;
+
+            RenderSystem.setShader(GameRenderer::getPositionTexShader);
+            RenderSystem.setShaderTexture(0, texture.resource());
+
+            Matrix4f matrix = poseStack.last().pose();
+
+            BufferBuilder bufferBuilder = Tesselator.getInstance().getBuilder();
+            bufferBuilder.begin(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_TEX);
+            bufferBuilder
+                    .vertex(matrix, (renderX - groovesWidth), (renderY + height + groovesHeight), 0)
+                    .uv((tx1) * uScale, (ty2) * vScale)
+                    .endVertex();
+            bufferBuilder
+                    .vertex(matrix, (renderX + width + groovesWidth), (renderY + height + groovesHeight), 0)
+                    .uv((tx2) * uScale, (ty2) * vScale)
+                    .endVertex();
+            bufferBuilder
+                    .vertex(matrix, (renderX + width + groovesWidth), (renderY - groovesHeight), 0)
+                    .uv((tx2) * uScale, (ty1) * vScale)
+                    .endVertex();
+            bufferBuilder
+                    .vertex(matrix, (renderX - groovesWidth), (renderY - groovesHeight), 0)
+                    .uv((tx1) * uScale, (ty1) * vScale)
+                    .endVertex();
+            bufferBuilder.end();
+            BufferUploader.end(bufferBuilder);
```

Comment:
What is the difference between this and RenderUtils#drawTexturedRect?

Reply:
This specifies `tx1` and `tx2`, which is not possible with `RenderUtils#drawTexturedRect` right now (outside of changing belived textureWidth and height)